### ISSUE TITLE
Winetricks does not update

### ISSUE THREAD

**Issue by rijnhard** [brezerk/q4wine#77]
Hi 

On: 
Linux mint 17.3 (ubuntu 14.04) 
QtWine 1.3 from technik ppa

when i try to update winetricks (inside qt4wine) I get this in terminal: 

``` bash
/bin/bash: /bin/sh -c "/usr/bin/wget http://kegel.com/wine/winetricks -O /home/rijnhard/.config/q4wine/winetricks && /bin/chmod +x /home/rijnhard/.config/q4wine/winetricks": No such file or directory
```

but if I run the commands manually in terminal it works, sort of. 

``` bash
# does not work
$ ./home/rijnhard/.config/q4wine/winetricks
bash: cd: /home/rijnhard/.config/q4wine/winetricks: Not a directory

# works
$ cd /home/rijnhard/.config/q4wine/
$ ./winetricks
```

**Comment by brezerk** [comment]
Thanks for the bug report.

Can you show output of this command please:

# ls -al /home/rijnhard/.config/q4wine/winetricks

Thanks.

**Comment by rijnhard** [comment]
Ok so i changed that. 
it wasn't downloading so I downloaded it manually and overwrote the original. 
So I'm not sure if that was a problem. but the manual download I did was owned by root, which I changed to my user. 

unfortunately I can't confirm the issue. but I will try to uninstall q4wine and reinstall it and do before and after

**Comment by brezerk** [comment]
Not sure too. The only thing I can do here: to add some checks to produce much more user-friendly error.

**Comment by brezerk** [comment]
This commit https://github.com/brezerk/q4wine/commit/c7cf08f7a147642afd85952f65db6a5722d6a495 adds few checks to produce more user-friendly messages if designated winetricks file or folder has wrong permissions.

Thanks for reporting.

Reopen if needed.

**Comment by rijnhard** [comment]
this has come back

```
/bin/bash: env WINEPREFIX='/home/rijnhard/.local/share/wineprefixes/Win10x86-stable' WINE='/opt/wine-stable/bin/wine' WINEDLLPATH='/opt/wine-stable/lib/wine' WINELOADER='/opt/wine-stable/bin/wine' WINESERVER='/opt/wine-stable/bin/wineserver' WINEARCH='win32' /bin/sh -c "/home/rijnhard/.config/q4wine/winetricks --no-isolate dotnet48": No such file or directory
```

the funny thing is the directory exists and is owned by my user, I manually installed the winetricks script again, but now i cant install any dll's

**Comment by rijnhard** [comment]
again if I run it manually it works

**Comment by brezerk** [comment]
hm...
it looks like for some reasons your bash is taking ENTIRE line as a single command.
i.e.:

```
brezerk@himera ~ $ 'make all good'
bash: make all good: command not found
```

instead of running `make` and sullying `all good` as an arguments:

```
brezerk@himera ~ $ make all good 
make: *** No rule to make target 'all'.  Stop.
```

this makes me think that it is something to do with terminal program on it's arguments you specified in q4wine settings.

can you send me yor ~/.config/q4wine/default.conf to my mail?

**Comment by brezerk** [comment]
yeah

you are using bash as gui terminal :) this is a bit unexpected. but anyway, let me see i can proved any workaround:

```
 71 [console]                                                                                                                                                                                                                               
 72 args=                                                                                                                                                                                                                                   
 73 bin=/bin/bash  
```

**Comment by brezerk** [comment]
on the other hand, it does not makes any seance to use `bash` as you won't be able to see the `winetricks` output b/c it would be running in background.

can you try to use `xterm` instead?

**Comment by rijnhard** [comment]
@brezerk did you mean to set the terminal as `xterm`?

this happens

```
/usr/bin/xterm: No absolute path found for shell: env WINEPREFIX='/home/rijnhard/.local/share/wineprefixes/Win10x86-stable' WINE='/opt/wine-stable/bin/wine' WINEDLLPATH='/opt/wine-stable/lib/wine' WINELOADER='/opt/wine-stable/bin/wine-preloader' WINESERVER='/opt/wine-stable/bin/wineserver' WINEARCH='win32' /bin/sh -c "/home/rijnhard/.config/q4wine/winetricks --no-isolate allcodecs"
```

**Comment by brezerk** [comment]
@rijnhard yes, you need to install it.
also, you need to supply `-c` as an option:

```
       -c        If the -c option is present, then commands are read from the first non-option argument command_string.  If there are arguments after the command_string, the first argument is assigned to $0 and any remaining argu‐
                 ments are assigned to the positional parameters.  The assignment to $0 sets the name of the shell, which is used in warning and error messages.
```

btw: you may try to use this option for bash too.

**Comment by rijnhard** [comment]
so my `xterm` no longer has a `-c` option. and doing it with bash lead to the output being hidden.

Playing around this is what worked, but this should REALLY be documented somewhere, if you can tell me where I won't mind adding it (if I have rights).

![Screenshot from 2020-02-12 08-05-08](https://user-images.githubusercontent.com/4290536/74332674-f2173480-4d9e-11ea-8720-99499e9bca74.png)

It should either have `xterm` as a dependency and be preconfigured for it, or detect the configured shell and use that with presets.

**Comment by brezerk** [comment]
hi @rijnhard 

i guess you don't  need extra '/bin/bash -c' in arguments.

As for documentation... It looks like it is a bit outdated: https://q4wine.brezblock.org.ua/documentation/en_us/11-settings.html#userutils

Any suggestions on what would be the best description for this fields?

Thanks

**Comment by kkomara** [comment]
I had the same problem in my Ubuntu 18.04 LTS.  I use "/usr/bin/gnome-terminal" for my console and I could not get Winetricks to work properly.  After I dug around and read these set of posts I figured it out.  In the version of gnome-terminal I am using allot of the older commands are depricated.  To get it to work on my system I had to add "--command "  to the arguments.  Not there is a blank space after the d in command.  This if VERY important and took quite a bit of trail and error to find.  Good Luck !
